pkg/operator/operator.go

```diff
@@ -176,8 +176,9 @@ func (optr *Operator) sync(key string) error {
 		glog.Errorf("Failed getting operator config: %v", err)
 		return err
 	}
-	if operatorConfig.Controllers.Provider == "None" {
-		glog.V(1).Info("`None` provider specified, reporting available")
+	switch p := operatorConfig.Controllers.Provider; p {
+	case "vSphere", "None":
```

Comment:
I think this would better expressed using the existing constants:

```diff
--- a/pkg/operator/operator.go
+++ b/pkg/operator/operator.go
@@ -176,8 +176,8 @@ func (optr *Operator) sync(key string) error {
                glog.Errorf("Failed getting operator config: %v", err)
                return err
        }
-       switch p := operatorConfig.Controllers.Provider; p {
-       case "vSphere", "None":
+       switch p := Provider(operatorConfig.Controllers.Provider); p {
+       case VSphereProvider, NoneProvider:
                glog.V(1).Infof("%q provider specified, reporting available", p)
                if err := optr.statusAvailable(); err != nil {
                        glog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
```

Reply:
I also experimented by removing the two cases that are handled to see if the existing unit tests failed - they did not. Can we add a unit test for the vSphere case?

Reply:
And same question again regarding "vSphere" versus "vsphere".

Reply:
let's use VSphere everywhere for consistency, we can clean up and indeed unexport all those const in a follow up

Reply:
> let's use VSphere everywhere for consistency, we can clean up and indeed unexport all those const in a follow up

So defined once as "VSphere" but referenced symbolically throughout. Agreed? Or am I missing the obvious?

Reply:
It seems odd to me that we would use mxed case for VSphere while all of the other providers use strictly lowercase: aws, libvirt, openstack, kubemark, baremetal, and none.

Reply:
> I also experimented by removing the two cases that are handled to see if the existing unit tests failed - they did not. Can we add a unit test for the vSphere case?

Sure. I'll write up some unit tests for the operator.

Reply:
Resolved by cd6ac6f...3844ccf